pyplanet/apps/contrib/jukebox/migrations/001_map_folder_change_public_to_visibility.py

```diff
@@ -0,0 +1,23 @@
+from peewee import RawQuery
+from playhouse.migrate import migrate, SchemaMigrator
+
+from ..models.mapfolder import MapFolder
+
+
+def upgrade(migrator: SchemaMigrator):
+	try:
+		query = RawQuery(MapFolder, """SELECT * FROM `mapfolder` WHERE public = 1""")
+		public_folders = query.execute()
+
+		migrate(
+			migrator.add_column(MapFolder._meta.db_table, 'visibility', MapFolder.visibility),
+			migrator.drop_column(MapFolder._meta.db_table, 'public')
+		)
+
+		MapFolder.update(visibility='public').where(MapFolder.id << [f.id for f in public_folders]).execute()
+	except Exception as e:
+		print(str(e))
+		print('Migration failed but silencing error!')
+
+def downgrade(migrator: SchemaMigrator):
+	pass
```

Comment:
@TheMaximum can you make a downgrade for this migration as well please?

Reply:
Implemented the downgrade, but unable to test it (at the moment). Could you have a look, @tomvlk?